File: Data Structures & Algorithms/linked-list-cycle-detection/submission-2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public boolean hasCycle(ListNode head) {
        Set<ListNode> visited = new HashSet<>();
        ListNode cur = head;
        while(cur != null){
            if(!visited.add(cur)){
                return true;
            }
            cur = cur.next;
        }
        return false;
    }
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-2.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func removeNthFromEnd(head *ListNode, n int) *ListNode {
    if head == nil {
		return head
	}
	length:=0
	cur := head
	for cur!=nil{
		 length++
		 cur = cur.Next 
	}
	if length==n{
		return head.Next
	}
	cur = head
	
	idx := -1
	for cur!=nil{
		idx++
		if idx == length-n-1 {
			cur.Next = cur.Next.Next
		}
		cur = cur.Next
	}
	return head
	
}
